src/handlers.cpp

```diff
@@ -1043,14 +1043,34 @@ ImpuRegDataTask::RequestType ImpuRegDataTask::request_type_from_body(std::string
   return ret;
 }
 
+std::string ImpuRegDataTask::server_name_from_body(std::string body)
+{
+  TRC_DEBUG("Determining server name from '%s'", body.c_str());
+
+  rapidjson::Document document;
+  document.Parse<0>(body.c_str());
+
+  if (!document.IsObject() ||
+      !document.HasMember("server_name") ||
+      !document["server_name"].IsString())
+  {
+    TRC_DEBUG("Did not receive valid JSON with a 'server_name' element");
+    return "";
+  }
+  else
+  {
+    return document["server_name"].GetString();
+  }
+}
+
 void ImpuRegDataTask::run()
 {
   const std::string prefix = "/impu/";
   std::string path = _req.full_path();
 
   _impu = path.substr(prefix.length(), path.find_first_of("/", prefix.length()) - prefix.length());
   _impi = _req.param("private_id");
-  _provided_server_name = _req.param("server_name");
+  _provided_server_name = server_name_from_body(_req.get_rx_body());
```

Comment:
It's weird to put this above the log on the line below and not include it in the log below. In fact, I imagine the debug logs for this section of code are an absolute mess.

server_name_from_body logs the HTTP body, then we have the log below with the private ID and public ID, and then on PUTs request_type_from_body logs the HTTP body again.

I think we should either log the HTTP body once or just log what we've gleaned from the HTTP body.